File: java/crackcode/src/crackcode/ch1_2.java
package crackcode;
/**
 * 1.2 Write code to reverse a C-Style String. 
 * (C-String means that “abcd” is represented as five characters, including the null character.)
 */       
public class ch1_2 {
    public static void reverse(char str[]){
        for(int i = 0; i < str.length/2; i++){
            char tmp = str[i];
            str[i] = str[str.length - 1 - i];
            str[str.length - 1 - i] = tmp;
        }
    }
    public static void main(String[] args) {
        char str[] = {'a','b','c','d'};
        reverse(str);
        System.out.println(str);
        
        char str1[] = {'a','b','c','d','e'};
        reverse(str1);
        System.out.println(str1);
    }

}

File: java/crackcode/c/ch1_2.c
#include <stdio.h>
void reverse(char* str){
	char* start = str;
	char* end = str;

	while(*end != (char)NULL){
		end++;
	}
	end--;

	while(start < end){
		char tmp = *start;
		*start = *end;
		*end = tmp;
		start++;
		end--;
	}
}

int main(int argc, char** argv){
	char str[] = "abcd";
	reverse(str);
	printf("%s\n", str);
	char str1[] = "abcde";
	reverse(str1);
	printf("%s\n", str1);
	return 0;
}